x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/rest/RestDeleteSecretsAction.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.fleet.rest;
+
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+import org.elasticsearch.xpack.fleet.action.DeleteSecretAction;
+import org.elasticsearch.xpack.fleet.action.DeleteSecretRequest;
+
+import java.io.IOException;
+import java.util.List;
+
+public class RestDeleteSecretsAction extends BaseRestHandler {
```

Comment:
If these APIs are supposed to be available in Serverless (which I assume they are) then they need to be annotated
```suggestion
@ServerlessScope(INTERNAL)
public class RestDeleteSecretsAction extends BaseRestHandler {
```

I assume they're `INTERNAL` - that is, only fleet-server & kibana are supposed to be able to access them, no external clients.